lib/kubeclient/informer.rb

```diff
@@ -88,6 +101,13 @@ def watch_to_update_cache
         @watching.each { |q| q << notice }
       end
       @logger&.info("watch restarted: #{stop_reason}")
+
+      # wake the waiter unless it's dead so it does not hang around
+      begin
+        @waiter.run
+      rescue ThreadError # rubocop:disable Lint/SuppressedException
+      end
+      @waiter.join
```

Comment:
Oh I think I see — previously, leftover waiter thread(s) could execute `@watcher.finish` at any time, possibly interrupting unrelated watchers from later tests, right?

But if we somehow (e.g. 'ERROR') come here early during `sleep(@reconcile_timeout)`, we might block for a long time — default 15min right?  
Here, what we're potentially blocking is the existing worker thread's loop, which won't be directly felt by the app but can cause significant gaps in the watching?

Reply:
it would have killed a random new waiter before (not really in tests since they all build their own informer)

added a `Thread.pass` to fix the race condition (don't think it really happens since the watch is a http request so it will take some time, but it's cheap so 🤷 )